nemo/core/neural_modules.py

```diff
@@ -162,6 +169,142 @@ def __is_of_allowed_type(self, var):
         # Well, seems that everything is ok.
         return True
 
+    def export_config(self, config_file, config_dir="~/data/configs"):
+        """
+            A function that exports module "configuration" (i.e. init parameters) to a YAML file.
+            Raises a ValueError exception in case then parameters coudn't be exported.
+
+            Args:
+                config_file: yml file name
+
+                config_dir: directory where the file will be stored (DEFAULT: ~/data/configs)
+        """
+        # Check if generic export will work.
+        if not self.__validate_params(self._init_params):
+            raise ValueError(
+                "Generic Module export cannot work as some of the values are not primitive types (string, int, float) "
+                F"or (lists of/dicts of) primitive types. Please implement your own custom `export_config()` and "
+                F"`import_config()` methods for your custom Module class."
+            )
+
+        # Greate an absolute path.
+        abs_path_file = path.join(path.expanduser(config_dir), config_file)
+
+        # Create the dictionary to be exported.
+        to_export = {}
+
+        # Get module "full specification".
+        module_full_spec = str(self.__module__) + "." + str(self.__class__.__qualname__)
+        module_class_name = type(self).__name__
+        # print(module_full_spec)
+
+        # Check whether module belongs to a collection.
+        spec_list = module_full_spec.split(".")
+
+        # Do not check Neural Modules from unit tests.
+        if spec_list[0] == "tests":
+            # Set collection variables.
+            collection_type = "tests"
+            collection_version = None
+        else:
+            # Check if component belongs to any collection
+            if len(spec_list) < 3 or (spec_list[0] != "nemo" and spec_list[1] != "collection"):
+                logging.warning(
+                    "Module `{}` does not belong to any collection. This won't be allowed in the next release.".format(
```

Comment:
Why? What about the common pytorch modules?

Reply:
They will be moved to CV collection and Common collection

Reply:
(not in this PR thou)